presto-tests/src/main/java/com/facebook/presto/tests/statistics/Metric.java

```diff
@@ -15,22 +15,42 @@
 
 import com.facebook.presto.cost.PlanNodeStatsEstimate;
 
-import java.util.function.Function;
+import java.util.Objects;
+import java.util.Optional;
 
-public enum Metric
+public abstract class Metric<T>
 {
-    OUTPUT_ROW_COUNT(PlanNodeStatsEstimate::getOutputRowCount),
-    OUTPUT_SIZE_BYTES(PlanNodeStatsEstimate::getOutputSizeInBytes);
+    public abstract Optional<T> getValueFromPlanNodeEstimate(PlanNodeStatsEstimate planNodeStatsEstimate, StatsContext statsContext);
 
-    private final Function<PlanNodeStatsEstimate, Double> extractor;
+    public abstract Optional<T> getValueFromAggregationQueryResult(Object value);
 
-    Metric(Function<PlanNodeStatsEstimate, Double> extractor)
+    public abstract String getComputingAggregationSql();
+
+    public abstract String getName();
```

Comment:
Maybe instead of this method `getName` we could enum `Metric.Type`. Then metric would a plain interface. No hacky equals/hashCode. What do you think?

Reply:
I removed the hack-code (hashCode, equals) and replaced `getName()` with plain `getString()`